src/native/minipal/thread.h

```diff
@@ -0,0 +1,86 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#ifndef HAVE_MINIPAL_THREAD_H
+#define HAVE_MINIPAL_THREAD_H
+
+#include <pthread.h>
+#include <stdint.h>
+
+#if defined(__linux__)
+#include <unistd.h>
+#include <sys/syscall.h>
+#elif defined(__FreeBSD__)
+#include <pthread_np.h>
+#elif defined(__NetBSD__)
+#include <lwp.h>
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * Get the current thread ID.
+ *
+ * @return The current thread ID as a size_t value.
+ */
+static inline size_t minipal_get_current_thread_id(void)
+{
+#ifdef __wasm
+    return 0;
+#else
+    static __thread size_t tid;
+
+    if (!tid)
+#if defined(__linux__)
+        tid = (size_t)syscall(SYS_gettid);
+#elif defined(__APPLE__)
+    {
+        uint64_t thread_id;
+        pthread_threadid_np(pthread_self(), &thread_id);
+        tid = (size_t)thread_id;  // Cast the uint64_t thread ID to size_t
```

Comment:
This would not have worked on 32-bit Apple platforms. Hopefully, there is not going to be 32-bit platform with 64-bit thread IDs in the future.

Reply:
We're C11 now.  Can use `static_assert`